components/post_view/post_list.jsx

```diff
@@ -440,194 +260,168 @@ export default class PostList extends React.PureComponent {
         } else {
             this.loadingPosts = false;
             if (this.mounted) {
-                this.setState({
-                    atEnd: !moreToLoad && this.props.posts.length < this.props.postVisibility,
+                const atEnd = !moreToLoad && this.props.posts.length < this.props.postVisibility;
+                const newState = {
+                    atEnd,
                     autoRetryEnable: true,
-                });
+                };
+
+                this.setState(newState);
             }
             if (!this.state.autoRetryEnable) {
                 this.autoRetriesCount = 0;
             }
         }
     }
 
-    handleScroll = () => {
-        if (this.scrollAnimationFrame) {
-            window.cancelAnimationFrame(this.scrollAnimationFrame);
-        }
-
-        this.scrollAnimationFrame = window.requestAnimationFrame(() => {
-            // Only count as user scroll if we've already performed our first load scroll
-            this.hasScrolled = this.hasScrolledToNewMessageSeparator || this.hasScrolledToFocusedPost;
-            const postList = this.postListRef.current;
-            const postListScrollTop = postList.scrollTop;
+    renderRow = ({itemId, style}) => {
+        return (
+            <div style={style}>
+                <PostListRow
+                    listId={itemId}
+                    channel={this.props.channel}
+                    shouldHighlight={itemId === this.props.focusedPostId}
+                    post={this.props.postsObjById[itemId]}
+                    loadMorePosts={this.loadMorePosts}
+                />
+            </div>
+        );
+    };
 
-            if (postList.scrollHeight === this.previousScrollHeight) {
-                this.atBottom = this.checkBottom();
-            }
+    itemKey = (index) => {
+        const {postListIds} = this.state;
+        return postListIds[index] ? postListIds[index] : index;
+    }
 
-            this.updateFloatingTimestamp();
+    onScroll = ({scrollDirection, scrollOffset, scrollUpdateWasRequested}) => {
+        const isNotLoadingPosts = !this.state.postsLoading && !this.loadingPosts;
+        const didUserScrollBackwards = scrollDirection === 'backward' && !scrollUpdateWasRequested;
+        const isOffsetWithInRange = scrollOffset < 1000 || window.scrollOfsetHelper; //initial offset will be 0 before the desired first scroll position
```

Comment:
Can't see reference to `window.scrollOfsetHelper`.  Also, there's a typo.

Reply:
Nice catch 👍 . I forgot about this